test/controllers/contests_controller_test.rb

```diff
@@ -274,6 +274,14 @@ class ContestsControllerTest < ActionController::TestCase
                     'Calling send_certificates does not do it immediately.'
   end
 
+  test 'show_results' do
+    test_abilities @c, :preview, [nil, :panitia, [:marker, create(:long_problem, contest: @c)]], [:marking_manager]
+    @c.update(start_time: Time.zone.now + 4.seconds,
```

Comment:
What is this block testing for?

Reply:
did you mean, this?

test 'show_results' do
    test_abilities @c, :preview, [nil, :panitia, [:marker, create(:long_problem, contest: @c)]], [:marking_manager]
    @c.update(start_time: Time.zone.now + 4.seconds,
    
can you more specify which block/line. here, i can't see where the block is, precisely.

Reply:
Oh yeah to be clear I'm talking about lines 279 - 282. Basically everything apart from `test_abilities`.

Reply:
hmm, idk what's wrong with my github, but there's only show line 277-279

![image](https://user-images.githubusercontent.com/56295230/127767796-fd85c569-3b53-42c3-891a-df7794ae55f0.png)

i guess, i want to make sure that show_results is invoked when contest is over. but, i guess i'm wrong, so just remove those lines?

Reply:
i mean 279-282

Reply:
Oh so I guess there are two things here:

1. Seems like you are testing `show_results` but then the `test_abilities` code is testing summary permissions. We should not combine both of them as they both test different things. If you want to test `show_results` I suggest splitting it to a different test.

   It is true that `:summary` is only used in relation to `show_contests`. However, I still feel like they should not be combined together. Especially considering that for other functions, the test_abilities function tests the ability that is exactly the same as the test name.

2. You are only asserting that `show_results` return 200 by going to `:show`. However, regardless of what happens you can always go to `ContestsController#show` and it will always return 200; it will just return something else (e.g. contest have not started, the contest problems, etc). If you want to test that it is specifically the results that are being shown, you can possibly check the contents of the instance variables (e.g. `@user_contests`).

Regarding the Github interface, you can go to the "Files changed" tab and scroll down to my comment to see the lines surrounding it. Then you should be able to see lines 279-282.